x-pack/plugins/security_solution_serverless/public/plugin.ts

```diff
@@ -40,6 +41,8 @@ export class SecuritySolutionServerlessPlugin
     setupDeps: SecuritySolutionServerlessPluginSetupDeps
   ): SecuritySolutionServerlessPluginSetup {
     registerUpsellings(setupDeps.securitySolution.upselling, this.config.productTypes);
+    setupDeps.securitySolution.setAppLinksSwitcher(projectAppLinksSwitcher);
```

Comment:
I like how it adjusts the links for serverless and have permission checked. I also like that we move all the serverless relevent flags to plugin contract, very well organised. 👍